stdlib/string.ml

```diff
@@ -202,6 +202,17 @@ type t = string
 let compare (x: t) (y: t) = Stdlib.compare x y
 external equal : string -> string -> bool = "caml_string_equal" [@@noalloc]
 
+let subrange ?(first = 0) ?last s =
+  let first = if first < 0 then 0 else first in
+  let max = length s - 1 in
+  let last = match last with
+    | None -> max
+    | Some l when l > max -> max
+    | Some l -> l
+  in
```

Comment:
This could be written a bit more compactly:
```suggestion
let subrange ?(first = 0) ?(last = max_int) s =
  let first = max 0 first in
  let last = min (length s - 1) last in
```
(or is there a performance concern with min/max?)

Reply:
Changed as suggested.